python.md

```diff
@@ -1,103 +1,408 @@
-    ",".join(list)     # returns a string with list elements seperated by comma
-
-### Dict
-
-    dict.keys()
-    dict.values()
-    "key" in dict    # let's say this returns False, then...
-    dict["key"]      # ...this raises KeyError
-    dict.get("key")  # ...this returns None
-    dict.setdefault("key", 1)
+#### String Methods
+
+``` python
+str[0:4]
+len(str)
+
+string.replace("-", " ")
+",".join(list)  # join separating each element by the string
+"hi {0}".format('j')
+str.find(",")
+str.index(",")  # same, but raises IndexError
+str.count(",")
+str.split(",")
+
+str.lower()
+str.upper()
+str.title()
+
+str.lstrip()
+str.rstrip()
+str.strip()
+
+str.islower()
+```
+
+### Math
+``` python
+print(5 % 2)    # 1 (modulo)
+print(5 / 2)    # 2.5 (division)
+print(5 // 2)   # 2.0 (int division)
+```
+    
+### Regex
+
+``` python
+import re
+
+re.match(r'^[aeiou]', str)
+re.sub(r'^[aeiou]', '?', str)
+re.sub(r'(xyz)', r'\1', str)
+
+expr = re.compile(r'^...$')
+expr.match(...)
+expr.sub(...)
+```
 
-### Iteration
 
-    for item in ["a", "b", "c"]:
-    for i in range(4):        # 0 to 3
-    for i in range(4, 8):     # 4 to 7
-    for i in range(1, 9, 2):  # 1, 3, 5, 7
-    for key, val in dict.items():
-    for index, item in enumerate(list):
+### Lists
+A list stores a series of items in a particular order. You
+access items using an index, or within a loop.  
+
+#### List Creation
+
+``` python
+list = []
+sites = ['github.com', 'devhints.io', 'python.org', ]
+```
+
+#### Get Items from List
+``` python
+list[i:j]  # returns list subset
+list[-1]   # returns last element
+list[:-1]  # returns all but the last element
+```
+
+### Modify items in a list
+
+``` python
+list[i] = val
+list[i:j] = otherlist  # replace ith to jth element with otherlist
+del list[i:j]
+```
+
+### List Methods
+
+``` python
+list.append(item)
+list.extend(another_list)
+list.insert(index, item)
+list.pop(i)       # returns and removes i-th element from the list
+list.remove(i)    # removes the first item from the list whose value is i
+list1 + list2     # combine two list    
+list(set(list))   # remove duplicate elements from a list
+
+list.reverse()    # reverses the elements of the list in-place
+list.count(item)
+sum(list)
 
-### [String](https://docs.python.org/2/library/stdtypes.html#string-methods)
+zip(list1, list2)  # returns list of tuples with n-th element of both list1 and list2
+list.sort()        # sorts in-place, returns None
 
-    str[0:4]
-    len(str)
+sorted(list)       # returns sorted copy of list
+",".join(list)     # returns a string with list elements seperated by comma
+```
 
-    string.replace("-", " ")
-    ",".join(list)
-    "hi {0}".format('j')
-    str.find(",")
-    str.index(",")   # same, but raises IndexError
-    str.count(",")
-    str.split(",")
+### Tuples
 
-    str.lower()
-    str.upper()
-    str.title()
+Tuples are similar to lists, but the items in a tuple can't be
+modified.
```

Comment:
The tuple can’t be modified, but the items in it can, so I think this line is misleading / imprecise.